File: src/screen.js
class Screen {
	constructor(left, top, width, height, colorDepth) {
		this._left = left;
		this._top = top;
		this._width = width;
		this._height = height;
		this._colorDepth = colorDepth;
	}

	get left() {
		return this._left;
	}
	get top() {
		return this._top;
	}
	get width() {
		return this._width;
	}

	get height() {
		return this._height;
	}

	get colorDepth() {
		return this._colorDepth;
	}

	fromJS(output) {
		output(this._width, this._height, this._colorDepth);
	}

	toJSON() {
		const {left, top, width, height, colorDepth} = this;
		return {left, top, width, height, colorDepth};
	}

	toString(output = this) {
		return `${output.left}x${output.top} ${output.width}x${output.height}x${output.colorDepth}`;
	}
}

module.exports = Screen;

File: src/screen.cc
#include "screen.h"

#include "nbind/nbind.h"

using namespace ScreenInfo;

Screen::Screen(size_t left, size_t top, size_t width, size_t height, int colorDepth) {
	this->left = left;
	this->top = top;
	this->width = width;
	this->height = height;
	this->colorDepth = colorDepth;
}

size_t Screen::getLeft() {
	return left;
}

size_t Screen::getTop() {
	return top;
}

size_t Screen::getWidth() {
	return width;
}

size_t Screen::getHeight() {
	return height;
}


int Screen::getColorDepth() {
	return colorDepth;
}

void Screen::toJS(nbind::cbOutput output) {
	output(left, top, width, height, colorDepth);
}



NBIND_CLASS(Screen) {
  construct<size_t, size_t, size_t, size_t, int>();
  method(getLeft);
  method(getTop);
  method(getWidth);
  method(getHeight);
  method(getColorDepth);
  method(main);
  method(all);
}


